tests/usbus_hid/main.c

```diff
@@ -0,0 +1,63 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "usb/usbus.h"
+#include "xtimer.h"
+#include "usb/usbus/hid.h"
+
+/*
+this descriptor is used, because the basic usb_hid interface was developed in conjunction with FIDO.
+Descriptor is taken from CTAP specification (version 20190130) section 8.1.8.2
+*/
+static uint8_t report_desc_ctap[] = {
+  0x06, 0xD0, 0xF1, /* HID_UsagePage ( FIDO_USAGE_PAGE ) */
+  0x09, 0x01,       /* HID_Usage ( FIDO_USAGE_CTAPHID ) */
+  0xA1, 0x01,       /* HID_Collection ( HID_Application ) */
+  0x09, 0x20,       /* HID_Usage ( FIDO_USAGE_DATA_IN ) */
+  0x15, 0x00,       /* HID_LogicalMin ( 0 ) */
+  0x26, 0xFF, 0x00, /* HID_LogicalMaxS ( 0xff ) */
+  0x75, 0x08,       /* HID_ReportSize ( 8 ) */
+  0x95, 0x40,       /* HID_ReportCount ( HID_INPUT_REPORT_BYTES ) */
+  0x81, 0x02,       /* HID_Input ( HID_Data | HID_Absolute | HID_Variable ) */
+  0x09, 0x21,       /* HID_Usage ( FIDO_USAGE_DATA_OUT ) */
+  0x15, 0x00,       /* HID_LogicalMin ( 0 ) */
+  0x26, 0xFF, 0x00, /* HID_LogicalMaxS ( 0xff ) */
+  0x75, 0x08,       /* HID_ReportSize ( 8 ) */
+  0x95, 0x40,       /* HID_ReportCount ( HID_OUTPUT_REPORT_BYTES ) */
+  0x91, 0x02,       /* HID_Output ( HID_Data | HID_Absolute | HID_Variable ) */
+  0xC0,             /* HID_EndCollection */
+};
+
+static usbus_t usbus;
+static char _stack[USBUS_STACKSIZE];
+
+void usb_hid_io_init(usbus_t* usbus, uint8_t* report_desc, size_t report_desc_size);
+ssize_t usb_hid_io_read(void* buffer, size_t len);
+
+void init(void)
```

Comment:
In RIOT this would typically be executed from `auto_init`. I assume that you implemented it here to pass `report_desc_ctap` correct? @bergzand do we have a similar situation with other USB classes?

Reply:
Correct. Because there are so many configuration options with HID report descriptors, I couldn't find a better solution than to simply disable auto_init_usbus and initialize it manually.